rest/response.go

```diff
@@ -92,6 +93,7 @@ func (f DefaultResponseFormatter) FormatList(ctx context.Context, headers http.H
 	}
 	if !skipBody {
 		payload := make([]map[string]interface{}, len(l.Items))
+		etags := make([]byte, len(l.Items)*md5.Size)
```

Comment:
Would be better to create a new md5 hash instance and append etags to it instead (incremental hash) instead of storing all the etags in memory to then create a hash.